modules/functions/src/admin/index.ts

```diff
@@ -0,0 +1,227 @@
+        description: 'List all functions',
+        queryParams: {
+          skip: ConduitNumber.Optional,
+          limit: ConduitNumber.Optional,
+          sort: ConduitString.Optional,
+        },
+      },
+      new ConduitRouteReturnDefinition('ListFunctions', 'String'),
+      this.listFunctions.bind(this),
+    );
+    this.routingManager.route(
+      {
+        path: '/:id',
+        action: ConduitRouteActions.GET,
+        description: 'Get a function',
+        urlParams: {
+          id: { type: RouteOptionType.String, required: true },
+        },
+      },
+      new ConduitRouteReturnDefinition('GetFunction', 'String'),
+      this.getFunction.bind(this),
+    );
+    this.routingManager.route(
+      {
+        path: '/executions',
+        action: ConduitRouteActions.GET,
+        description: 'List all functions executions',
+        queryParams: {
+          skip: ConduitNumber.Optional,
+          limit: ConduitNumber.Optional,
+          sort: ConduitString.Optional,
+        },
+      },
+      new ConduitRouteReturnDefinition('ListFunctionsExecutions', 'String'),
+      this.getFunctionsExecutions.bind(this),
+    );
+    this.routingManager.route(
+      {
+        path: '/executions/:functionName',
+        action: ConduitRouteActions.GET,
+        description: 'Get function executions for specific function',
+        urlParams: {
+          functionName: { type: RouteOptionType.String, required: true },
+        },
+        queryParams: {
+          success: ConduitBoolean.Optional,
+        },
+      },
+      new ConduitRouteReturnDefinition('GetFunctionExecutions', 'String'),
+      this.getFunctionExecutions.bind(this),
+    );
+    this.routingManager.route(
+      {
+        path: '/:id',
+        action: ConduitRouteActions.PATCH,
+        description: 'Update a function',
+        urlParams: {
+          id: { type: RouteOptionType.String, required: true },
+        },
+        bodyParams: {
+          name: ConduitString.Optional,
+          functionCode: ConduitString.Optional,
+          inputs: { type: TYPE.JSON, required: false },
+          returns: { type: TYPE.JSON, required: false },
+          timeout: ConduitNumber.Optional,
+        },
+      },
+      new ConduitRouteReturnDefinition('UpdateFunction', 'String'),
+      this.updateFunction.bind(this),
+    );
+    this.routingManager.registerRoutes();
+  }
+
+  async uploadFunction(call: ParsedRouterRequest): Promise<UnparsedRouterResponse> {
+    const { name, functionCode, inputs, returns } = call.request.params;
+    const func = await Functions.getInstance().findOne({ name: name });
+    if (!isNil(func)) {
+      throw new GrpcError(status.ALREADY_EXISTS, 'function name already exists');
+    }
+    const timeoutValue = inputs.timeout ?? 180000;
+    const query = {
+      name,
+      functionCode,
+      inputs,
+      returns,
+      timeout: timeoutValue,
+    };
+    return Functions.getInstance().create(query);
```

Comment:
How is the new function going to be published? You only create the model, you do not refresh the routes